### ISSUE TITLE
Brackets not working; instead adds new measure

### ISSUE THREAD

**Issue by TimBomBom** [musescore/MuseScore#24060]
### Issue type

UX/interaction bug

### Description with steps to reproduce

I tried adding brackets around an accidental the other day to remind the player that the note was a natural, not flat, but I was unable to put brackets around the natural sign. In fact, I could not add brackets to any element at all. I made a test file and saw that attempting to add brackets instead creates a new measure for some reason.

### Supporting files, videos and screenshots

https://github.com/user-attachments/assets/95e03279-7b5d-4df1-a4f2-70de1d52c743

### What is the latest version of MuseScore Studio where this issue is present?

MuseScore Studio version (64-bit): 4.3.2-241630831, revision: github-musescore-musescore-22b46f2

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

OS: Windows 10 Version 2009 or later, Arch.: x86_64

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by oktophonie** [comment]
Another appearance of the mysterious https://github.com/musescore/MuseScore/issues/22959

**Comment by TimBomBom** [comment]
> Another appearance of the mysterious #22959

Interesting. I searched for relevant issues but couldn't find it. Guess it's because I was searching for "brackets" whereas this was parantheses

**Comment by zacjansheski** [comment]
I still haven't been able to reproduce this problem, strange!
We now have examples from both Mac and Windows users

Could you think of anything special you did when adding the parenthesis to the palette?
Does resetting and re-adding the parenthesis to the palette solve the issue?

It might be worth checking if the issue is still present in the new Beta https://musescore.org/en/node/367021

**Comment by cbjeukendrup** [comment]
You could upload the diagnostic files (in MuseScore, do Diagnostic > Save diagnostic files, choose a location, and upload the resulting zip file here). That should also contain the file that contains your palettes definitions. That won't tell us how it got into this state, but may give some insight of what the state _is_.

**Comment by TimBomBom** [comment]
I managed to fix it by deleting and re-adding the brackets to my accidentals tab! I'll still attach my diagnostic files from before I attempted though just in case you can find the root cause.

[diagnostic.zip](https://github.com/user-attachments/files/16641713/diagnostic.zip)

**Comment by cbjeukendrup** [comment]
Thanks for uploading! The workspace file indeed immediately explains the problem. This is the code that corresponds to the palette item you are using:
```xml
    <Cell name="Add parentheses to element">
      <ActionIcon>
        <subtype>21</subtype>
        <action>add-parentheses</action>
        </ActionIcon>
      </Cell>
```

The `action` is `add-parentheses`, and which icon you see is based on this action. So you see indeed an "add parentheses" icon.

But the `subtype` is 21, and apparently what this element really does is determined by this "subtype". When we look at [the `ActionIconType` enum](https://github.com/musescore/MuseScore/blob/a6e6c7cc9970419e6b25a0e5638018616377b244/src/engraving/dom/actionicon.h#L29) as it is in the latest MuseScore version, we see that a value of 21 corresponds to the `MEASURE` type, while the `PARENTHESES` type corresponds to a value of 22. 

This has been caused by the insertion of the `STAFF_TYPE_CHANGE` type in 4aacacfbef871118e3343448af74e5ab3f50089c. Since the insertion of that type, the numbers of all types below it have been increased by one, but this is not being considered when reading palette files that were saved by a previous version of MuseScore.

The easy fix would have been to add compatibility reading code just at the moment that 4aacacfbef871118e3343448af74e5ab3f50089c was merged: if that were done immediately, the version number written to the metadata of the palette file could have been used to decide how to interpret the numbers. But that's too late now: since the file was already re-saved using MuseScore 4.3.2, the version number is now 4.3.2, instead of the version number of the last release before the change.

What we can do though, is stop using the `<subtype>` field entirely, and determine everything based on the `<action>` field. That will still fix it even for people who already used 4.3.2 or any other version between that change and now. 

It looks like "the change" was first included in MuseScore 4.1.0. So everyone who has used MuseScore 4.0.x _and_ customised their palettes using that version so that a palette file gets written (_and_ never reset to factory settings in between) should experience this bug. 

Let this be a reminder for all software developers in the world: never serialise enum values by converting them to integers and writing those integers to a file, but always use some meaningful textual key.
